### ISSUE TITLE
[Bug][a11y]: Button when used with href renders without any ARIA role

### ISSUE THREAD

**Issue by dineshy87** [adobe/spectrum-web-components#3245]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Button

### Expected behavior

When used with href attribute, change the role of the component to link. 

Note: This variant is not mentioned in the [button api documentation](https://opensource.adobe.com/spectrum-web-components/components/button/api/) but is available as example on [storybook](https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/story/button-accent-fill--link). If this is being used by some team, or added as an option but not documented, then it needs to be fixed and documented. If this isn't meant to be supported, then please remove it from storybook so that it's not used by any product teams.

### Actual behavior

When href is added to sp-button, a link with aria-hidden is rendered. And at the same time, role is also missing on the component. So effectively, the interactive nature of component is not communicated to screen reader users.

### How can we reproduce this issue?

1. Go to 'https://opensource.adobe.com/spectrum-web-components/storybook/index.html?path=/story/button-accent-fill--link'
2. Use voiceover or any other screenreader and check what is being announced.
3. It is announced as group(voiceover) or as clickable(NVDA) instead of as a **link**. (NVDA announces any element that has  mousedown events as clickable if they don't have any other roles like button or link as a fallback)

**Comment by najikahalsema** [comment]
The solution is twofold:

1. Adding documentation for an href button is the first step and pretty straightforward. 
2. Adding the role to an href button is already being done within `manageAnchor()` in `ButtonBase.ts`, but perhaps the logic for this role management is flawed.

**Comment by Rajdeepc** [comment]
PR fix: https://github.com/adobe/spectrum-web-components/pull/3329

**Comment by Rajdeepc** [comment]
@dineshy87 Can you also review this from your side and let us know?

**Comment by dineshy87** [comment]
This works.

**Comment by Rajdeepc** [comment]
Closing as this is verified by the reported. Thanks
